routes/status.js

```diff
@@ -17,16 +17,45 @@ const DAYS = 24 * HOURS;
 const WEEKS = 7 * DAYS;
 const MONTH = 30 * DAYS;
 const YEARS = 365 * DAYS;
+const NOT_EQUAL = 1;
+const EQUAL = 0;
 
 async function get_status(req, res, next, request_query = {}, limit = 0) {
 	let _res;
 	let where_sql = "";
 	let sql_arr = [];
 	for (let i in request_query) {
-		if (typeof request_query[i] !== "undefined" && typeof request_query[i] !== "boolean") {
+		if (typeof request_query[i] === "string") {
```

Comment:
```suggestion
		if (typeof request_query[i] === "string" || typeof request_query[i] === "int") {
```
```javascript
	const problem_id = req.params.problem_id === "null" ? undefined : parseInt(req.params.problem_id);
	const user_id = req.params.user_id === "null" ? undefined : req.params.user_id;
	const language = req.params.language === "null" ? undefined : parseInt(req.params.language);
	const result = req.params.result === "null" ? undefined : parseInt(req.params.result);
```
`problem_id`, `language` and `result` is `int` not `string`, this code will **NOT** apply filter.